### ISSUE TITLE
$verbosePreference = 'Continue' not affect Stop-Process cmdlet (and maybe others cmdlets)

### ISSUE THREAD

**Issue by carloscondit** [PowerShell/PowerShell#19225]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

The [help](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_preference_variables?view=powershell-7.3#verbosepreference) says that $VerbosePreference should affect scripts, cmdlets and providers. When setting the $VerbosePreference variable to Continue, the detailed output of the Stop-Process cmdlet does not appear, although it should, considering if you specify the Verbose parameter specifically, the verbose output is shown.
Perhaps there are other cmdlets whose detailed output is not affected by $VerbosePreference = 'Continue'.

### Expected behavior

```console
PS C:\Temp> $VerbosePreference = 'Continue'
PS C:\Temp> notepad.exe
PS C:\Temp> Stop-Process -name notepad
VERBOSE: Performing the operation "Stop-Process" on target "Notepad (18904)".
PS C:\Temp>
```

### Actual behavior

```console
PS C:\Temp> $VerbosePreference = 'Continue'
PS C:\Temp> notepad.exe
PS C:\Temp> Stop-Process -name notepad
PS C:\Temp> 
PS C:\Temp> notepad.exe
PS C:\Temp> Stop-Process -name notepad -Verbose
VERBOSE: Performing the operation "Stop-Process" on target "Notepad (3528)".
```

### Environment data

```powershell
PS C:\Temp> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.3
PSEdition                      Core
GitCommitId                    7.3.3
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Temp>
```

### Visuals

![Verbose](https://user-images.githubusercontent.com/39201353/221399086-88ad663d-2d7d-411d-b3a7-6349741f057c.png)

**Comment by carloscondit** [comment]
It was explained to me that preferences can be overwritten by the developer of the cmdlet. I think that this information should be added to the help in the Verbose Preference section.

**Comment by mklement0** [comment]
I'd say that you've discovered a _bug_, and a very long-standing one at that: It goes back to Windows PowerShell.

Only _some_ cmdlets do not honor the preference-variable settings - such as `Stop-Process` and `Remove-Item`; which ones do not should be investigated systematically and fixed.

Others _do_, such as `Invoke-RestMethod`; in the following example, only `Invoke-RestMethod`'s verbose output prints:

```powershell
& { 
  $VerbosePreference = 'Continue'
  ''> temp:/test.txt
  Remove-Item temp:/test.txt
  Invoke-RestMethod -Method HEAD http://example.org 
}
```

**Comment by carloscondit** [comment]
@mklement0 @iSazonov
Apparently there is already an open issue on this. https://github.com/PowerShell/PowerShell/issues/12148
Right?

**Comment by mklement0** [comment]
Thanks, @jhoneill. I'm not aware of a list, I just stumbled upon the differences.

However, perhaps looking at the implementation of two cmdlets that behave differently in that regard (e.g., `Stop-Process` vs. `Invoke-RestMethod`) will reveal a pattern that allows systematic discovery of the cmdlets that are broken in this respect.

**Comment by jhoneill** [comment]
@mklement0  OK I'll raise it. The more examples we have the easier tracking down any pattern(s) would be.

**Comment by SteveL-MSFT** [comment]
Looking at `Stop-Process`, it seems this is exactly due to https://github.com/PowerShell/PowerShell/issues/12148.  Details of investigation added there.

**Comment by MartinGC94** [comment]
> **YES** I can suggest this in the cmdlet working group, but I would prefer not to keep going back with another and then another is there a list?

There are 2 different classes you can inherit from when creating cmdlets: "cmdlet" and "pscmdlet". Stop-Process is using "cmdlet" so presumably every command that is using that is affected. You can create your own list with this script:
```
# Import every module so ImplementingType is populated
Get-Module -ListAvailable | group Name | foreach {
    $_.Group | Sort Version -Descending | select -First 1
} | Import-Module

Get-Command -CommandType Cmdlet | Where {
    $null -ne $_.ImplementingType -and
    !$_.ImplementingType.IsSubclassOf([System.Management.Automation.PSCmdlet])
}
```

Here's the list from my computer:
<details>
  <summary>Commands</summary>
  

  ```
Add-KdsRootKey
Add-RegKeyProperty
Clear-KdsCache
Clear-Tpm
Confirm-SecureBootUEFI
ConvertFrom-Json
ConvertTo-TpmOwnerAuth
Copy-RegKey
Copy-UserInternationalSettingsToSystem
Debug-Process
Disable-AppBackgroundTaskDiagnosticLog
Disable-LocalUser
Disable-TlsSessionTicketKey
Disable-TpmAutoProvisioning
Dismount-RegHive
Enable-AppBackgroundTaskDiagnosticLog
Enable-LocalUser
Enable-TlsSessionTicketKey
Enable-TpmAutoProvisioning
Format-SecureBootUEFI
Get-CertificateAutoEnrollmentPolicy
Get-CertificateEnrollmentPolicyServer
Get-CertificateNotificationTask
Get-CimAssociatedInstance
Get-CimClass
Get-CimInstance
Get-CimSession
Get-DAPolicyChange
Get-Date
Get-InstalledLanguage
Get-KdsRootKey
Get-LocalGroup
Get-LocalGroupMember
Get-LocalUser
Get-MouseSettings
Get-Process
Get-RegKey
Get-SecureBootPolicy
Get-SecureBootUEFI
Get-Service
Get-SystemPreferredUILanguage
Get-Tpm
Get-TpmEndorsementKeyInfo
Get-TpmSupportedFeature
Get-Verb
Get-WheaMemoryPolicy
Get-WinAcceptLanguageFromLanguageListOptOut
Get-WinCultureFromLanguageListOptOut
Get-WinDefaultInputMethodOverride
Get-WindowsDeveloperLicense
Get-WinHomeLocation
Get-WinLanguageBarOption
Get-WinSystemLocale
Get-WinUILanguageOverride
Get-WinUserLanguageList
Import-TpmOwnerAuth
Initialize-Tpm
Invoke-CimMethod
New-CimInstance
New-CimSession
New-CimSessionOption
New-Guid
New-LocalGroup
New-NetIPsecAuthProposal
New-NetIPsecMainModeCryptoProposal
New-NetIPsecQuickModeCryptoProposal
New-RegKey
New-Service
New-TemporaryFile
New-WinUserLanguageList
Remove-CertificateEnrollmentPolicyServer
Remove-CertificateNotificationTask
Remove-CimInstance
Remove-CimSession
Remove-LocalGroup
Remove-LocalUser
Remove-RegKey
Remove-RegKeyProperty
Remove-Service
Rename-LocalGroup
Rename-LocalUser
Rename-RegKey
Resolve-DnsName
Restart-Service
Resume-Service
Set-AppBackgroundTaskResourcePolicy
Set-CertificateAutoEnrollmentPolicy
Set-CimInstance
Set-Culture
Set-LocalGroup
Set-SecureBootUEFI
Set-Service
Set-SystemPreferredUILanguage
Set-TpmOwnerAuth
Set-WheaMemoryPolicy
Set-WinAcceptLanguageFromLanguageListOptOut
Set-WinCultureFromLanguageListOptOut
Set-WinDefaultInputMethodOverride
Set-WinHomeLocation
Set-WinLanguageBarOption
Set-WinSystemLocale
Set-WinUILanguageOverride
Set-WinUserLanguageList
Show-WindowsDeveloperLicenseRegistration
Start-Service
Stop-Process
Stop-Service
Suspend-Service
Test-KdsRootKey
Unblock-Tpm
Uninstall-Language
Unregister-WindowsDeveloperLicense
Wait-Process
  ```
</details>

**Comment by jhoneill** [comment]
@MartinGC94  That's not an especially helpful list - some things like Set-Culture don't send output even when -Verbose is specified... Plus quite a few of those are not part of PowerShell -e.g. the TPM ones. I'll leave others to comment on whether that presumption is valid

**Comment by SteveL-MSFT** [comment]
Dupe of https://github.com/PowerShell/PowerShell/issues/12148.  For other cmdlets that deal with Verbose directly incorrectly, please open a new issue.
